elastic/kibana
# [Profiling] Fix default value of profiling settings

Fixes: https://github.com/elastic/kibana/issues/177592

## Summary

This PR changes the format of the default value for the `profilingAWSCostDiscountRate` and the `profilingAzureCostDiscountRate` settings from a string to a number. Since the `type` property is not specified in these setting definitions, the advanced settings components assume the type of the settings based on the type of the default value. Since the default value was specified as a string, the settings components used a text field instead of a number field and so the validation did not work properly.

The PR also adds a small fix to the settings normalization function where, if the provided `value` property in the setting definition is `0`, it is incorrectly translated to `undefined` rather than `0`.

**How to test:**
1. Go to Stack Management -> Advanced settings
2. Verify that the initial value in the fields is `0` rather than empty fields.
3. Set an invalid value (<0 or >100) for these two settings and verify that an error message is displayed.


<img width="1335" alt="Screenshot 2024-02-23 at 13 53 04" src="https://github.com/elastic/kibana/assets/59341489/9b6f191b-ae17-4f46-9814-2a325d083fd6">

<img width="1335" alt="Screenshot 2024-02-23 at 13 41 14" src="https://github.com/elastic/kibana/assets/59341489/95fe4685-4504-44c9-b312-9530dcbe1b2b">
